internal/policy/policy.go

```diff
@@ -523,7 +400,69 @@ func (s *State) Verify(ctx context.Context) error {
 		return err
 	}
 
-	return targetsVerifier.Verify(ctx, nil, s.TargetsEnvelope)
+	if err := targetsVerifier.Verify(ctx, nil, s.TargetsEnvelope); err != nil {
+		return err
+	}
+
+	targetsMetadata, err := s.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return err
+	}
+
+	reachedDelegations := map[string]bool{}
+	for delegatedRoleName := range s.DelegationEnvelopes {
+		reachedDelegations[delegatedRoleName] = false
+	}
+
+	delegationsQueue := targetsMetadata.Delegations.Roles
+	delegationKeys := targetsMetadata.Delegations.Keys
+	for {
+		if len(delegationsQueue) <= 1 {
+			break
```

Comment:
Added a comment. This is because the last rule is always the allow rule. We don't actually process it during our search, but it's there explicitly to indicate to the user how rules work.